app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt

```diff
@@ -113,16 +115,17 @@ class MonthView(context: Context, attrs: AttributeSet, defStyle: Int) : View(con
     }
 
     private fun groupAllEvents() {
-        days.forEach {
-            val day = it
-            day.dayEvents.forEach {
-                val event = it
+        days.forEach { day ->
 
+            day.dayEvents.forEach { event ->
                 // make sure we properly handle events lasting multiple days and repeating ones
                 val lastEvent = allEvents.lastOrNull { it.id == event.id }
                 val daysCnt = getEventLastingDaysCount(event)
                 val validDayEvent = isDayValid(event, day.code)
-                if ((lastEvent == null || lastEvent.startDayIndex + daysCnt <= day.indexOnMonthView) && !validDayEvent) {
+
+                val lastDay = { findLastDay(event) ?: day }
```

Comment:
do we need the brackets here?

Reply:
ah, this is supposed to be a function? Make it a proper function then please, this is unreadable

Reply:
Yes, it's a lambda

Without the lambda, the `findLastDay()` (iterates through the whole `days` array and events) will be executed needlessly in each loop even when there are no repeating events 

I created the lambda for readability only, the `findLastDay() :? day` term could be moved to the if statement itself avoiding the need for a lambda

How about local functions? Allowed?

Reply:
that sounds really bad, just add some casual function please

Reply:
https://github.com/SimpleMobileTools/Simple-Calendar/pull/1722/commits/ac55c060d51f53723ae43fad7955d7cd2cfe6675

(the nullability check was unnecessary)